### ISSUE TITLE
bug(wasm): Origo proving freezes main Chrome Extension UI

### ISSUE THREAD

**Issue by 0xFloyd** [pluto/web-prover#364]
### tldr; Origo proving freezes main UI even though its running in the same codepath as TLSN. is there a need for browser specific Web APIs like window/document to be referenced in the wasm for anything? Is it a library dependency, or wasm-bindgen? If we do need it, fine we look for another solution. If we don't we can move proving in the extension to its own thread and solve all problems

For reference, it appears that the extension UI/ main thread seems to freeze sometime after the log client::origo_wasm32: js_computed_witnesses

```
2025-01-09T22:23:20.899Z INFO client::origo_wasm32: js_computed_witnesses:
```

UI appears to freeze somewhere after the js_computed_witnesses step, but all the logs below pop up at the same time, so not sure which exactly is the issue. The thread appears to be blocked until the proof is successfully generated

```
2025-01-09T22:23:22.691Z DEBUG client::origo_wasm32: initializing public params
2025-01-09T22:23:22.693Z DEBUG proofs::program::data: loading proving params, proving_param_bytes=115598577
2025-01-09T22:23:25.094Z DEBUG proofs::program::data: done loading proving params
2025-01-09T22:23:25.094Z INFO proofs::program::data: circuit list
2025-01-09T22:23:31.695Z INFO proofs::program::data: circuit shapes
2025-01-09T22:23:36.015Z INFO proofs::program::data: public params from parts
```

### Background: 
Origo proving (same with TLSN proving) must be done in the chrome extension offscreen document context due to some wasm library reliance on Browser specific Web APIs like window and document. Offscreen documents run on the main thread, which freezes the main chrome extension UI. Proving, and also witness generation, can’t be moved to its own thread in a worker due this reliance on Browser Web APIs which don’t exist in a Worker thread context.

Additionally, we are unable to compile some cryptography libraries to wasm. Because of this, we must generate the witness outside wasm in the chrome extension. 

Proof generation in the chrome extension has quite a few limitations. The biggest one being wasm execution must occur within a browser environment where certain Web APIs like the document object is available, as the document object is referenced somewhere internally in the wasm by some library (currently unknown where/what in the wasm is referencing this).

This is the reason in the extension we run the wasm in a chrome [Offscreen Document ](https://developer.chrome.com/blog/Offscreen-Documents-in-Manifest-v3), as offscreen documents are a browser context where document exists. 

This solved our initial problem of generating a proof with TLSN. However, with Origo proving, (and also witness generation), there is heavy computation and/or libraries with large overhead. We are able to successfully generate a witness and create an Origo proof in the extension. However, it appears that the offscreen document runs in the same single-threaded JavaScript environment associated with the main extension sidebar UI (I've been unable to find documentation on this) as the UI is frozen when generating an Origo proof (that's pretty good confirmation of sharing the same thread). The [JavaScript Event Loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Event_loop) explains how heavy computations can block the UI if done on the main thread.

There is an obvious solution to this which we attempted to implement: move Origo proving to a [Worker](https://developer.mozilla.org/en-US/docs/Web/API/Worker) - this would offload the execution to its own thread. However, certain Web APIs are [not available in Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Functions_and_classes_available_to_workers), and the document object is one of them.

For witness calc, both [poseidon-lite](https://www.npmjs.com/package/poseidon-lite) and [snarkjs](https://www.npmjs.com/package/snarkjs) require execution within a browser environment as they reference the document object. Additionally, the witness must be made available on the `window.witness` namespace, as the wasm [references this](https://github.com/pluto/web-prover/blob/c244626075ce6778e04921562cf378bb8b7235c0/client/src/origo_wasm32.rs#L58) and needs this to exist. 

```rust
#[wasm_bindgen]
extern "C" {
  #[wasm_bindgen(js_namespace = witness, js_name = createWitness)]
  async fn create_witness_js(input: &JsValue) -> JsValue;
}
```

I think we must figure out a solution that allows Origo proving to be executed in its own thread
